simphony/cuds/lattice_new.py

```diff
@@ -0,0 +1,342 @@
+
+        Parameters
+        ----------
+        nodes : iterable of LatticeNode objects
+            reference to LatticeNode objects from where the data is copied
+            to the Lattice
+
+        """
+        for node in nodes:
+            index = node.index
+            if any(value < 0 for value in index):
+                raise IndexError('invalid index: {}'.format(index))
+            self._dcs[index] = DataContainer(node.data)
+
+    def iter_nodes(self, indices=None):
+        """Get an iterator over the LatticeNodes described by the indices.
+
+        Parameters
+        ----------
+        indices : iterable set of int[3], optional
+            When indices (i.e. node index coordinates) are provided, then nodes
+            are returned in the same order of the provided indices. If indices
+            is None, there is no restriction on the order the nodes that are
+            returned.
+
+        Returns
+        -------
+        A generator for LatticeNode objects
+
+        """
+        if indices is None:
+            for index, val in np.ndenumerate(self._dcs):
+                yield self.get_node(index)
+        else:
+            for index in indices:
+                yield self.get_node(index)
+
+    def count_of(self, item_type):
+        """ Return the count of item_type in the container.
+
+        Parameters
+        ----------
+        item_type : CUDSItem
+            The CUDSItem enum of the type of the items to return the count of.
+
+        Returns
+        -------
+        count : int
+            The number of items of item_type in the container.
+
+        Raises
+        ------
+        ValueError :
+            If the type of the item is not supported in the current
+            container.
+
+        """
+        try:
+            return np.prod(self._items_count[item_type]())
+        except KeyError:
+            error_str = "Trying to obtain count a of non-supported item: {}"
+            raise ValueError(error_str.format(item_type))
+
+    @property
+    def size(self):
+        return self._size
+
+    @property
+    def origin(self):
+        return self._origin
+
+    @property
+    def prim_cell(self):
+        return self._prim_cell
+
+    @property
+    def data(self):
+        return DataContainer(self._data)
+
+    @data.setter
+    def data(self, value):
+        self._data = DataContainer(value)
+
+
+def make_cubic_lattice(name, h, size, org=(0, 0, 0)):
+    """Create and return a 3D (primitive) cubic lattice.
+
+    Parameters
+    ----------
+    name : str
+    h : float
+        lattice spacing
+    size : int[3]
+        Number of lattice nodes in each axis direction.
+    org : float[3], default value = (0, 0, 0)
+        lattice origin
+
+    Returns
+    -------
+    lattice : Lattice
+        A reference to a Lattice object.
+    """
+    pcell = pc.create_cell_cubic_lattice(h)
+    return Lattice(name, 'Cubic', size, org, pcell)
```

Comment:
I think that these need to be updated with the new constructor of the Lattice class, there is no need to pass the lattice kind anymore